### ISSUE TITLE
Failure cause is difficult to determine on test-integration

### ISSUE THREAD

**Issue by enj** [kubernetes/kubernetes#69106]
(sorry if this is a dupe)

On https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/69044/pull-kubernetes-integration/28676/ the failure is reported as:

```
Test Failures
test-cmd run_recursive_resources_tests 9.06s

go run hack/e2e.go -v --test --test_args='--ginkgo.focus=test\-cmd\srun\_recursive\_resources\_tests$'

error: error validating "hack/testdata/recursive/rc/rc/busybox-broken.yaml": error validating data: kind not set; if you choose to ignore these errors, turn validation off with --validate=false
error: error validating "hack/testdata/recursive/deployment/deployment/nginx-broken.yaml": error validating data: kind not set; if you choose to ignore these errors, turn validation off with --validate=false
warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.
error: unable to decode "hack/testdata/recursive/deployment/deployment/nginx-broken.yaml": Object 'Kind' is missing in '{"apiVersion":"extensions/v1beta1","ind":"Deployment","metadata":{"labels":{"app":"nginx2-deployment"},"name":"nginx2-deployment"},"spec":{"replicas":2,"template":{"metadata":{"labels":{"app":"nginx2"}},"spec":{"containers":[{"image":"k8s.gcr.io/nginx:1.7.9","name":"nginx","ports":[{"containerPort":80}]}]}}}}'
error: error validating "hack/testdata/recursive/rc/rc/busybox-broken.yaml": error validating data: kind not set; if you choose to ignore these errors, turn validation off with --validate=false
!!! [0926 01:04:33] Call tree:
!!! [0926 01:04:33]  1: /go/src/k8s.io/kubernetes/test/cmd/../../third_party/forked/shell2junit/sh2ju.sh:47 run_recursive_resources_tests(...)
!!! [0926 01:04:33]  2: /go/src/k8s.io/kubernetes/test/cmd/../../third_party/forked/shell2junit/sh2ju.sh:112 eVal(...)
!!! [0926 01:04:33]  3: /go/src/k8s.io/kubernetes/test/cmd/legacy-script.sh:133 juLog(...)
!!! [0926 01:04:33]  4: /go/src/k8s.io/kubernetes/test/cmd/legacy-script.sh:525 record_command(...)
!!! [0926 01:04:33]  5: hack/make-rules/test-cmd.sh:109 runTests(...)
				
error: error validating "hack/testdata/recursive/rc/rc/busybox-broken.yaml": error validating data: kind not set; if you choose to ignore these errors, turn validation off with --validate=false
error: error validating "hack/testdata/recursive/deployment/deployment/nginx-broken.yaml": error validating data: kind not set; if you choose to ignore these errors, turn validation off with --validate=false
warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.
error: unable to decode "hack/testdata/recursive/deployment/deployment/nginx-broken.yaml": Object 'Kind' is missing in '{"apiVersion":"extensions/v1beta1","ind":"Deployment","metadata":{"labels":{"app":"nginx2-deployment"},"name":"nginx2-deployment"},"spec":{"replicas":2,"template":{"metadata":{"labels":{"app":"nginx2"}},"spec":{"containers":[{"image":"k8s.gcr.io/nginx:1.7.9","name":"nginx","ports":[{"containerPort":80}]}]}}}}'
error: error validating "hack/testdata/recursive/rc/rc/busybox-broken.yaml": error validating data: kind not set; if you choose to ignore these errors, turn validation off with --validate=false
!!! [0926 01:04:33] Call tree:
!!! [0926 01:04:33]  1: /go/src/k8s.io/kubernetes/test/cmd/../../third_party/forked/shell2junit/sh2ju.sh:47 run_recursive_resources_tests(...)
!!! [0926 01:04:33]  2: /go/src/k8s.io/kubernetes/test/cmd/../../third_party/forked/shell2junit/sh2ju.sh:112 eVal(...)
!!! [0926 01:04:33]  3: /go/src/k8s.io/kubernetes/test/cmd/legacy-script.sh:133 juLog(...)
!!! [0926 01:04:33]  4: /go/src/k8s.io/kubernetes/test/cmd/legacy-script.sh:525 record_command(...)
!!! [0926 01:04:33]  5: hack/make-rules/test-cmd.sh:109 runTests(...)

				from junit_test-cmd.xml

Filter through log files | View test history on testgrid 
```

After some effort sifting through I was able to determine the actual failure:

```
I0926 01:04:32.215] has:Object 'Kind' is missing
I0926 01:04:32.215] deployment.extensions "nginx1-deployment" force deleted
I0926 01:04:32.216] deployment.extensions "nginx0-deployment" force deleted
I0926 01:04:33.204] generic-resources.sh:424: Successful get rc {{range.items}}{{.metadata.name}}:{{end}}: 
I0926 01:04:33.345] (Breplicationcontroller/busybox0 created
I0926 01:04:33.348] replicationcontroller/busybox1 created
I0926 01:04:33.443] generic-resources.sh:428: Successful get rc {{range.items}}{{.metadata.name}}:{{end}}: busybox0:busybox1:
I0926 01:04:33.531] (BFAIL!
I0926 01:04:33.532] message:no rollbacker has been implemented for "ReplicationController"
I0926 01:04:33.532] no rollbacker has been implemented for "ReplicationController"
I0926 01:04:33.532] unable to decode "hack/testdata/recursive/rc/rc/busybox-broken.yaml": Object 'Kind' is missing in '{"apiVersion":"v1","ind":"ReplicationController","metadata":{"labels":{"app":"busybox2"},"name":"busybox2"},"spec":{"replicas":1,"selector":{"app":"busybox2"},"template":{"metadata":{"labels":{"app":"busybox2"},"name":"busybox2"},"spec":{"containers":[{"command":["sleep","3600"],"image":"busybox","imagePullPolicy":"IfNotPresent","name":"busybox"}],"restartPolicy":"Always"}}}}'
I0926 01:04:33.532] has not:no rollbacker has been implemented for {"" "ReplicationController"}
I0926 01:04:33.533] 433 /go/src/k8s.io/kubernetes/test/cmd/../../test/cmd/generic-resources.sh
I0926 01:04:33.569] +++ exit code: 1
I0926 01:04:33.575] +++ error: 1
I0926 01:04:33.606] Error when running run_recursive_resources_tests
```

The important part being:

```
I0926 01:04:33.532] has not:no rollbacker has been implemented for {"" "ReplicationController"}
I0926 01:04:33.533] 433 /go/src/k8s.io/kubernetes/test/cmd/../../test/cmd/generic-resources.sh
```

It would be really helpful if the failure message told me to look at `test/cmd/generic-resources.sh:433` without having to dig through the build log.

xref: https://github.com/kubernetes/test-infra/issues/9585 #69044 
/sig testing

**Comment by BenTheElder** [comment]
> It would be really helpful if the failure message told me to look at test/cmd/generic-resources.sh:433 without having to dig through the build log.

This should be fairly straightforward if anyone wants to take it up

/help
/good-first-issue

**Comment by aruneli** [comment]
I will take this issue @BenTheElder

**Comment by taragu** [comment]
@aruneli is this still up for grabs? if yes i'd like to take a stab at it

**Comment by taragu** [comment]
Closed by #70619

/close

**Comment by serathius** [comment]
Can someone rewrite it from bash to golang?
